doc/modules/tree.rst

```diff
@@ -334,9 +334,7 @@ total cost over the entire trees (by summing the cost at each node) of
 Tips on practical use
 =====================
 
-  * Decision trees tend to overfit on data with a large number of features.
```

Comment:
The VC dimension aside, I am under the impression that both statements are correct and provide 2 different pieces of information.

For the former sentence, for a fixed `n_samples`, increasing `n_features` allows the tree to find additional splits to isolate noisy samples and therefore increases its capacity to overfit.

The proposed statement is linked to the capacity of the decision tree to perfectly learn a training set if no limits are given regarding the depth.

The VC dimension will just tell you the complexity of the model and therefore you can compare the complexity of two models for the two scenarios. But both trees are going to overfit.

To summarize, I think that both statements are complementary.